quality_of_life_changer.client/src/redux-modules/user-profile/sagas.js

```diff
@@ -0,0 +1,50 @@
+import { put, takeEvery } from "redux-saga/effects";
+import { userProfileConstants } from "./constants";
+import * as actions from "./actions";
+import fetcher from "../../utils/fetcher";
+import { toastr } from "react-redux-toastr";
+import { constants } from "../../constants/constants";
+
+function* fetchUserCalendars({ id }) {
+  try {
+    const { data } = yield fetcher.get(
+      "api/user/",
+      { id },
+      "/user-profile/calendars"
+    );
+    toastr.success("Your calendars loaded");
+    yield put(actions.setCalendars(data.calendars));
+  } catch (error) {
+    console.log(error);
+    toastr.error("Error", error.message);
+  }
+}
+
+function* fetchCalendarStatus({ name, url }) {
+  try {
+    const token = localStorage.getItem(constants.JWT_TOKEN);
+    const tokenPayload = JSON.parse(window.atob(token.split(".")[1]));
+    const id = tokenPayload.nameid;
+    const { data } = yield fetcher.post(
+      "api/user/",
```

Comment:
please use string interpolation here, like `api/user/${id}/user-profile/add-calendar`
also, i have a question about this URL. make sure it meets RESTful API url principles